test/unit/carriers/usps_test.rb

```diff
@@ -451,6 +451,30 @@ def test_extract_event_details_handles_double_digit_calendar_dates
     assert_equal 12, details.zoneless_time.mday
   end
 
+  def test_correct_login_succeeds_in_valid_credentials?
+    @carrier = USPS.new(:login => '677JADED7283')
+    @carrier.expects(:test_mode?).returns(true)
+    assert @carrier.valid_credentials?
+  end
+
+  def test_wrong_login_fails_in_valid_credentials?
+    @carrier = USPS.new(:login => 'ABCDEFGHIJKL')
+    @carrier.expects(:test_mode?).returns(true)
+    refute @carrier.valid_credentials?
+  end
+
+  def test_blank_login_fails_in_valid_credentials?
+    @carrier = USPS.new(:login => '')
+    @carrier.expects(:test_mode?).returns(true)
+    refute @carrier.valid_credentials?
```

Comment:
Given that you only use the variable locally, no need to use an instance variable. SO you can ditch the `@` in front of `@carrier`.